cpp/include/raft/neighbors/detail/ivf_pq_search.cuh

```diff
@@ -1379,10 +1377,11 @@ void ivfpq_search_worker(raft::device_resources const& handle,
   rmm::device_uvector<float> query_kths(0, stream, mr);
   if (manage_local_topk) {
     query_kths.resize(n_queries, stream);
-    thrust::fill_n(handle.get_thrust_policy(),
-                   query_kths.data(),
-                   n_queries,
-                   float(dummy_block_sort_t<ScoreT, IdxT>::queue_t::kDummy));
+    linalg::detail::map<false>(
+      stream,
+      query_kths.data(),
+      n_queries,
```

Comment:
I wonder if this is an opportunity to expose our own "fill_n" function. Or would that just be too redundant to map?

Reply:
Technically, we have it in [matrix/init.cuh](https://github.com/rapidsai/raft/blob/1c321f298a0d308efd0795512221fb1093904aad/cpp/include/raft/matrix/init.cuh#L64-L70). However, here use the `detail` version because the input comes in `rmm::uvector`. I cannot use the mdspan api here, because I resize it conditional on runtime arguments ("enable" the buffer only if it's used). 

Perhaps, this would be a good place to use `std::optional<mdarray>`, but it's hard to `emplace` the optional when we mostly use helpers to create mdarrays.

Reply:
To your last point- would `std::make_optional` not work here? The `mdarray` is moveable as far as I know.

Reply:
I guess it could work (didn't try yet), but may be clumsy to force specify all template parameters.

Reply:
Update: turned out not so bad, although it didn't reduce the number of lines.

What do you think about backporting [`std::optional::transform`](https://en.cppreference.com/w/cpp/utility/optional/transform) from C++23 into raft? I miss this function so much for one-line constructs like this:
```C++
query_kths.transform([](auto x){return x.data_handle();}).value_or(std::nullptr);
```